### DISCUSSION TITLE
Preparing for the first sprint demo

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#41] [Archive]
We're about half way through the first week of the first real sprint and we need a solid plan for our first demo.

This will be as much an introduction to the concept of AaC as a working software demonstration.  I think showing some examples will be key.  I recommend we look at the existing models folder as our demonstration cases and adjust those as appropriate for our demo.  I expect the demo needs to be time boxed to about 15 min or so, so we'll have to be concise.  Alternatively we can do an introduction to AaC the day before the demo to get the audience familiarized with the concept.

Draft Demo Flow:

1. Review the README
2. Install:  pip install aac
3. Show command line usage:  aac --help
3. Show AaC is self-validating:  aac validate .\model\aac\AaC.yaml
3. Install: pip install aac-plantuml
4. Tour a model...recommend Flow because it is more interesting than the others
5. Validate flow system spec:  aac validate .\model\flow\System.yaml
6. Generate some PlantUML (component, object)
7. Validate flow use case:  aac validate .\model\flow\Flow.yaml
8. Generate PlantUML sequence diagram:  aac puml-sequence .\model\flow\Flow.yaml
9. Show the pipeline, highlight the automated tests and quality checks
10. Mention the plugin capabilty - probably defer demonstration till later (after we've templatized things)

We can use the PlantUML web site to show the outputs converted into graphics for the demo.

The objective of the demo should be to prepare for some early user engagement and feedback in the following sprint.  One of our other teams seems interested in testing things out for us on a real system specification.  It would be great if we can be prepared to make that happen.  We may need to add documentation to our demo if we can get it prioritized and built out.

Any feedback on the demo flow?  Do you think we need to carve out some more time?

**Comment by cvchaparro** [comment]
I think we should plan allot the demo 30 minutes - 10-15 minutes of which would be our demo and the rest of the time we could spend answering questions, etc. Of course, seeing as how there seems to be interest already, 15-20 minutes for this may not be enough time but it would be a start.

As far as the flow for the demo, it seems like a logical approach. Assuming Alex and I would be responsible for the demo, I think it would be best if we just have one person do the whole thing (especially since it's a short demo, anyways) and everyone could contribute to answering questions but I could also see a reason to split the demo amongst everyone, if that's preferable.

**Comment by Coffee2Bits** [comment]
I like the demo flow; I don't have any changes that I'd suggest. I think the time will be fine, but Cameron's observation about initial questions taking extra time is a good call-out.  

What kind of demo documentation would we want to create? A general runbook/demo steps, or did you have something more detailed in mind?

**Comment by jondavid-black** [comment]
I talked to Thomas today and he said they should be able to just hit the highlights and give us some sprint demo time.  Let's plan for 20-30 min...including Q&A.

**Comment by jondavid-black** [comment]
I'd really like to get things moved into the Jellyfish repo before the demo on Friday.  Do you guys both have working e-mails so I can get you added to the GitHub org?

**Reply by Coffee2Bits** [reply]
We have. I'll send you an additional message about this.

**Comment by jondavid-black** [comment]
Another consideration...when is the right time to blow away Jellyfish 2.0?  I'm thinking sooner is better since I'm not aware of any active users.

**Reply by Coffee2Bits** [reply]
I think immediately is fine. I agree, I don't think there are any active users.

**Comment by jondavid-black** [comment]
Tomorrow is that last day to wrap things up for the demo.  Have you run through the steps of the demo and shown everything works?  Are there any last minute updates we think are critical to include?

**Comment by Coffee2Bits** [comment]
I did a preliminary walkthrough today and aside from a minor issue with plantuml (I already have a fix) the main branch is good to demo, but my plan is to spend more time tomorrow refining/timing the demo.

The plugin command line args would be a good item to show off extensions with since it has both enum and data extensions in a demonstrable way. I am working through a validator issue with data extensions at the moment, but I expect to have the PR up tomorrow.

All of that to say I think this demo agenda is good, and if my timing of the demo tomorrow is too short then we have some more things we can easily add. 👍

**Reply by jondavid-black** [reply]
Well...the demo starts at 8:30.  Hope we're ready.

**Comment by Coffee2Bits** [comment]
I noticed that the other teams presenting were referencing their JIRA boards to step through the new features developed during the sprint--would we want to do something similar with the github issues? We could use milestones for grouping issues in sprints (especially since they have due dates).

**Reply by cvchaparro** [reply]
I think that's a good idea as it could give some visibility (i.e. show our priorities, etc) into what tasks we were specifically working on for the sprint for people that aren't actively following what we're doing.

**Comment by jondavid-black** [comment]
Good job on the first demo!  I didn't really want to spend too much time staring at GitHub just yet with the team.

**Comment by Coffee2Bits** [comment]
I think we can lock/archive this discussion unless there are some demo improvements that we'd like to see.

**Comment by lizzcondrey** [comment]
This is marked as resolved and sprint is well past. Going to close this discussion.
